File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-java/src/main/java/com/example/statemachine/simple/doument/config/DocumentIdChecker.java
package com.example.statemachine.simple.doument.config;

import com.example.statemachine.simple.doument.config.exception.PersistenceDataIsNotExistException;
import java.util.Objects;
import org.springframework.jdbc.core.JdbcTemplate;

public class DocumentIdChecker {
  private final JdbcTemplate jdbcTemplate;

  public DocumentIdChecker(JdbcTemplate jdbcTemplate) {
    this.jdbcTemplate = jdbcTemplate;
  }

  public boolean isExist(final int documentId) throws PersistenceDataIsNotExistException {
    final var cnt = Objects.requireNonNull(
      jdbcTemplate.queryForObject("SELECT COUNT(id) FROM documents WHERE id = ?",
        Integer.class, documentId));

    if (cnt != 1) {
      final var message = String.format(
        "주어진 documentId \'%s\' 는 데이터베이스에 존재하지 않는 값입니다", documentId);
      throw new PersistenceDataIsNotExistException(message);
    }

    return true;
  }
}

File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-kotlin/src/main/kotlin/com/example/statemachine/simple/document/config/DocumentIdChecker.kt
package com.example.statemachine.simple.document.config

import com.example.statemachine.simple.document.config.exception.PersistenceDataIsNotExistException
import org.springframework.jdbc.core.JdbcTemplate

class DocumentIdChecker(private val jdbcTemplate: JdbcTemplate) {

  fun isExist(documentId: Int): Boolean {
    val cnt = jdbcTemplate.queryForObject(
      "SELECT COUNT(id) AS cnt FROM documents WHERE id = ?",
      Int::class.java,
      documentId
    ).toInt()


    if (cnt != 1) {
      throw PersistenceDataIsNotExistException(
        "주어진 documentId '$documentId' 는 데이터베이스에 존재하지 않는 값입니다"
      )
    }
    return true
  }
}
